sdks/java/core/src/main/java/org/apache/beam/sdk/util/RowFilter.java

```diff
@@ -0,0 +1,347 @@
+    this.rowSchema = rowSchema;
+  }
+
+  /**
+   * Checks whether a {@link Schema} contains a list of field names. Nested fields can be expressed
+   * with dot-notation. Throws a helpful error in the case where a field doesn't exist, or if a
+   * nested field could not be reached.
+   */
+  @VisibleForTesting
+  static void validateSchemaContainsFields(
+      Schema schema, List<String> specifiedFields, String operation) {
+    Set<String> notFound = new HashSet<>();
+    Set<String> notRowField = new HashSet<>();
+
+    for (String field : specifiedFields) {
+      List<String> levels = Splitter.on(".").splitToList(field);
+
+      Schema currentSchema = schema;
+
+      for (int i = 0; i < levels.size(); i++) {
+        String currentFieldName = String.join(".", levels.subList(0, i + 1));
+
+        if (!currentSchema.hasField(levels.get(i))) {
+          notFound.add(currentFieldName);
+          break;
+        }
+
+        if (i + 1 < levels.size()) {
+          Schema.Field nextField = currentSchema.getField(levels.get(i));
+          if (!nextField.getType().getTypeName().equals(Schema.TypeName.ROW)) {
+            notRowField.add(currentFieldName);
+            break;
+          }
+          currentSchema = Preconditions.checkNotNull(nextField.getType().getRowSchema());
+        }
+      }
+    }
+
+    if (!notFound.isEmpty() || !notRowField.isEmpty()) {
+      String message = "Validation failed for " + operation + ".";
+      if (!notFound.isEmpty()) {
+        message += "\nRow Schema does not contain the following specified fields: " + notFound;
+      }
+      if (!notRowField.isEmpty()) {
+        message +=
+            "\nThe following specified fields are not of type Row. Their nested fields could not be reached: "
+                + notRowField;
+      }
+      throw new IllegalArgumentException(message);
+    }
+  }
+
+  /**
+   * Configures this {@link RowFilter} to filter {@link Row}s by keeping only the specified fields.
+   * Nested fields can be specified using dot-notation.
+   *
+   * <p>For example, if we want to keep the list of fields {@code ["foo", "baz.nested_1"]}, for the
+   * input {@link Row}:
+   *
+   * <pre>{@code
+   * foo: 123
+   * bar: 456
+   * baz:
+   *   nested_1: abc
+   *   nested_2: xyz
+   * }</pre>
+   *
+   * we will get the following output {@link Row}:
+   *
+   * <pre>{@code
+   * foo: 123
+   * baz
+   *   nested_1: abc
+   * }</pre>
+   */
+  public RowFilter keeping(List<String> fields) {
+    Preconditions.checkState(
+        transformedSchema == null,
+        "This RowFilter has already been configured to filter to the following Schema: %s",
+        transformedSchema);
+    validateSchemaContainsFields(rowSchema, fields, "\"keep\"");
+    transformedSchema = keepFields(rowSchema, fields);
+    return this;
+  }
+
+  /**
+   * Configures this {@link RowFilter} to filter {@link Row} by removing the specified fields.
+   * Nested fields can be specified using dot-notation.
```

Comment:
Do we have a clear usecase for this? I'm not convinced it's worth the complexity (not just here, but for other transforms (possibly in other languages) that might also adopt these conventions, and further other tools that might want to do things like validation).

Reply:
This is about reaching nested fields?

No one asked for it AFAIK, just thought it would be a nice-to-have. 

Beam already somewhat has this convention: our [`Select`](https://github.com/apache/beam/blob/2a0ce8f910363377cd57ca759df8daf476789688/sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java#L100) transform (that does a similar "drop" operation on Beam Rows) also reaches nested fields